service/matching/service.go

```diff
@@ -22,76 +22,71 @@ package matching
 
 import (
 	"github.com/uber/cadence/common"
-	"github.com/uber/cadence/common/log/loggerimpl"
+	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
-	"github.com/uber/cadence/common/persistence/client"
+	"github.com/uber/cadence/common/persistence"
+	persistenceClient "github.com/uber/cadence/common/persistence/client"
+	"github.com/uber/cadence/common/resource"
 	"github.com/uber/cadence/common/service"
 	"github.com/uber/cadence/common/service/dynamicconfig"
 )
 
 // Service represents the cadence-matching service
 type Service struct {
-	stopC  chan struct{}
-	params *service.BootstrapParams
+	resource.Resource
 	config *Config
+
+	stopC chan struct{}
 }
 
 // NewService builds a new cadence-matching service
-func NewService(params *service.BootstrapParams) common.Daemon {
-	config := NewConfig(dynamicconfig.NewCollection(params.DynamicConfig, params.Logger))
-	params.ThrottledLogger = loggerimpl.NewThrottledLogger(params.Logger, config.ThrottledLogRPS)
-	params.UpdateLoggerWithServiceName(common.MatchingServiceName)
-	return &Service{
-		params: params,
-		config: config,
-		stopC:  make(chan struct{}),
+func NewService(
+	params *service.BootstrapParams,
+) (resource.Resource, error) {
+
+	serviceConfig := NewConfig(dynamicconfig.NewCollection(params.DynamicConfig, params.Logger))
+	serviceResource, err := resource.New(
+		params,
+		common.MatchingServiceName,
+		serviceConfig.ThrottledLogRPS,
+		func(
```

Comment:
Do you really need this func? Seems like all you are doing is calling persistenceBean.GetVisibilityManager()

Reply:
frontend / history will have some overrides